lib/permit_ecto/resolver.ex

```diff
@@ -17,6 +17,39 @@ defmodule Permit.Ecto.Resolver do
   import Permit.Helpers, only: [resource_module_from_resource: 1]
 
   @impl Permit.ResolverBase
+  def resolve(
+        subject,
+        authorization_module,
+        resource_module,
+        action,
+        %{use_loader?: true} = meta,
+        _
+      ) do
+    params_for_loader = %{
+      action: action,
+      resource_module: resource_module,
+      subject: subject,
+      params: meta[:params]
+    }
+
+    with {_, true} <-
+           {:pre_auth, authorized?(subject, authorization_module, resource_module, action)},
+         resource <- meta[:loader].(params_for_loader),
+         {_, true} <-
+           check_authorized(resource, subject, authorization_module, action) do
+      {:authorized, resource}
+    else
+      {:pre_auth, false} ->
+        :unauthorized
+
+      {:auth, false} ->
+        :unauthorized
+
+      nil ->
+        raise Ecto.NoResultsError
```

Comment:
Since it's a library that depends on Ecto, this is acceptable, but it's possible that it could have a different behaviour since we're not actually firing any query explicitly here. (Related to #1)